apps/store/themes/store/js/categorization.js

```diff
@@ -0,0 +1,261 @@
+                            url = removeURLParameter(decodeURIComponent(window.location.href),
+                                data, true);
+                            if(url.split("q=")[1] !== ""){
+                                url = url + "%2C" + encodedQueryString;
+                            } else {
+                                url = url + encodedQueryString;
+                            }
+                        } else {
+                            url = window.location.href + "%2C" + encodedQueryString;
+                        }
+                    } else {
+                        url = window.location.href + encodedQueryString;
+                    }
+                } else {
+                    url = removeURLParameter(decodeURIComponent(window.location.href), data, false);
+                }
+
+            } else {
+                url = caramel.tenantedUrl('/assets/' + store.asset.type + '/list?' + buildParams(encodedQueryString));
+            }
+
+            loadAssets(url);
+        }
+
+        $('.search-bar h2').find('.page').text(' / Search: "' + searchQuery + '"');
+    };
+
+    var buildParams = function (query) {
+        return 'q=' + query;
+    };
+
+    var getEncodedQueryString = function (searchQuery) {
+        var q = {};
+        var output = '';
+        if(searchQuery !== ''){
+            q = parseUsedDefinedQuery(searchQuery);
+            q = JSON.stringify(q);
+            q = q.replace('{','').replace('}', '');
+            q = encodeURIComponent(q);
+            output =q;
+        }
+        return output;
+    };
+
+    var triggerEvent = function (data, isRemove){
+        var searchQueryString = data.parent + ":" + data.text;
+
+        if((window.location.href.indexOf(data.parent) > -1) && !isRemove){
+            searchQueryString = updateORQuery(window.location.href, data);
+        }
+        search(searchQueryString, data, isRemove);
+    };
+
+    var updateORQuery = function (url, data) {
+        var value = getParamValue(url, data.parent).trim();
+        var updatedQuery = "";
+        if(value.indexOf("(") > -1){
+            updatedQuery = data.parent + ":(" + value.substring(value.indexOf("\"(")+2, value.indexOf(")\"")) + " OR "
+                + data.text + ")";
+        } else {
+            value = "\"" + value.split("\"").join("").trim() + "\"";
+            updatedQuery = data.parent + ":(" + value.replace("\"", "").replace("\"", "") + " OR " + data.text + ")";
+        }
+        return updatedQuery;
+    };
+
+    var removeURLParameter = function(sourceURL, data, removeWhole) {
+        var rtn = sourceURL.split("?")[0],
+            param,
+            params_arr = [],
+            queryString = (sourceURL.indexOf("?") !== -1) ? sourceURL.split("?")[1] : "";
+        if (queryString !== "") {
+            params_arr = queryString.split("&");
+            for (var i = params_arr.length - 1; i >= 0; i -= 1) {
```

Comment:
These nested for loops have been refactored already.

Please add a comment explaining the logic.